internal/agent/device/bootstrap.go

```diff
@@ -110,6 +119,31 @@ func (b *Bootstrap) Initialize(ctx context.Context) error {
 	return nil
 }
 
+func (b *Bootstrap) ensureCurrentRenderedSpecUpToDate(ctx context.Context) error {
+	desiredSpec, err := spec.ReadRenderedSpecFromFile(b.deviceReader, b.desiredRenderedFile)
+	if err != nil {
+		if err == spec.ErrMissingRenderedSpec {
+			return nil
+		}
+		return fmt.Errorf("getting current rendered spec: %w", err)
+	}
+
+	bootcCmd := container.NewBootcCmd(b.executer)
+	bootcHost, err := bootcCmd.Status(ctx)
+	if err != nil {
+		return fmt.Errorf("getting current bootc status: %w", err)
+	}
+
+	if container.IsOsImageReconciled(bootcHost, &desiredSpec) {
+		err = spec.WriteRenderedSpecToFile(b.deviceWriter, &desiredSpec, b.currentRenderedFile)
```

Comment:
lets log this as info

Reply:
we want container.IsOsImageReconciled and something like below that signals spec is stale
```
func IsOsImageInTransition(current *v1alpha1.RenderedDeviceSpec, desired * v1alpha1.RenderedDeviceSpec ) bool {
 return current.Spec.Os.Image != desired.Spec.Os.Image
}
```

Reply:
lets log a  `Warn`  `if container.IsOsImageReconciled` is not true. this means the device was rebooted before  Apply in which case it will reconcile later. Or worse we failed to upgrade. But right now we have not understanding of what to do so we continue.

Reply:
Done